compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -11653,6 +11653,41 @@ TR::Node *sxorSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    return node;
    }
 
+//---------------------------------------------------------------------
+// Floating point square root
+//
+
+TR::Node *fdsqrtSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
```

Comment:
Is there really enough shared code in this function to justify sending both float and double opcodes here? It seems much simpler to me to just have a fsqrtSimplifier and a dsqrtSimplifier unless you anticipate augmenting this function with lots of folding that doesn't care about the type. This might even be one of those cases where I wouldn't be averse to using a template <gasp>.

Reply:
Good suggestion, I think template is the best solution in this case. Let me update it.
About the shared code, I agree currently it doesn't have too much common path between float and double; but in the future we may have flag based transformations such as transforming sqrt negative child into NaN. Template will work better in this situation as well.

Reply:
Changes have been updated to use the template approach.